nxt_editor/stage_model.py

```diff
@@ -3146,6 +3190,24 @@ def process_events():
         QtCore.QCoreApplication.processEvents()
 
 
+class NxtUndoStack(QtWidgets.QUndoStack):
+
+    def push(self, command):
+        """Simple overload of push method, checks that the target layer of the given command's model is *not* locked.
+        If the command does not have a model attr nothing is checked.
+
+        :param command: Command to push to undo stack
+        :type command: QUndoCommand
+        :return: None
+        """
+        model = getattr(command, 'model', None)  # type: StageModel
+        if model and model.target_layer.get_locked():
+            logger.error('The target layer is locked!')
+            model.request_ding.emit()
```

Comment:
Is this an `error` or a `warning`? In one sense we're telling them they made an error, but in another all we're doing is warning them nothing happened. There's been no internal error. I don't think the codebase is particularly consistent on this.

Much more importantly, let's put the ding behind a preference. Either one here to be specific about attempting to edit a locked layer, or a more general one in the `ding` function to silence the ding.